dev/com.ibm.ws.kernel.boot.core/src/com/ibm/ws/kernel/internal/classloader/DirectoryResourceHandler.java

```diff
@@ -73,6 +76,33 @@ public Manifest getManifest() throws IOException {
         return manifest;
     }
 
+    @Override
+    public Set<String> getClassPackages() {
+        Set<String> packages = new HashSet<>();
+        StringBuilder sb = new StringBuilder();
+        getPackages(directory, packages, sb);
```

Comment:
Right.  It is only called once, so is there any concern here?  I could move the getClassPackages() function to the JarFileClassLoader instead of in the ResourceHandler methods if that would be better.

Reply:
Just bringing it to you attention in case you hadn't noticed because the method has public access.  I had considered the method is called once in this solution, so didn't request a change here.  I like the current organization of the getClassPackages() behavior.  If usage changes we can always modify the method to return the previously computed set.